### ISSUE TITLE
borrowed data escapes outside of method after `cargo clippy --fix`

### ISSUE THREAD

**Issue by trueb2** [rust-lang/rust-clippy#15615]
### Summary

```
    Checking qsib-client v0.1.0 (/Users/jwtrueb/Desktop/workspace/qsib-fetcher/qsib-client)
warning: failed to automatically apply fixes suggested by rustc to crate `qsib_client`

after fixes were automatically applied the compiler reported errors within these files:

  * qsib-client/src/client.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0521]: borrowed data escapes outside of method
   --> qsib-client/src/client.rs:503:26
    |
491 |         queries: &[Query],
    |         -------  - let's call the lifetime of this reference `'1`
    |         |
    |         `queries` is a reference that is only valid in the method body
...
503 |             let handle = tokio::spawn(async move { client.download(q, out).await });
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                          |
    |                          `queries` escapes the method body here
    |                          argument requires that `'1` must outlive `'static`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0521`.
Original diagnostics will follow.

warning: redundant redefinition of a binding `start_ms`
   --> qsib-client/src/client.rs:284:17
    |
284 |                 let start_ms = start_ms;
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^
    |
help: `start_ms` is initially defined here
   --> qsib-client/src/client.rs:256:13
    |
256 |         let start_ms = query.start_time.timestamp_millis();
    |             ^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_locals
    = note: `#[warn(clippy::redundant_locals)]` on by default

warning: redundant redefinition of a binding `end_ms`
   --> qsib-client/src/client.rs:285:17
    |
285 |                 let end_ms = end_ms;
    |                 ^^^^^^^^^^^^^^^^^^^^
    |
help: `end_ms` is initially defined here
   --> qsib-client/src/client.rs:257:13
    |
257 |         let end_ms = query.end_time.timestamp_millis();
    |             ^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_locals

warning: used consecutive `str::replace` call
   --> qsib-client/src/client.rs:291:26
    |
291 |                           .replace(' ', "_")
    |  __________________________^
292 | |                         .replace('(', "_")
293 | |                         .replace(')', "_")
294 | |                         .replace('.', "_")
295 | |                         .replace(':', "_")
296 | |                         .replace('/', "_")
297 | |                         .replace('\\', "_");
    | |___________________________________________^ help: replace with: `replace([' ', '(', ')', '.', ':', '/', '\\'], "_")`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_str_replace
    = note: `#[warn(clippy::collapsible_str_replace)]` on by default

warning: unnecessary use of `cloned`
   --> qsib-client/src/client.rs:506:18
    |
506 |         for q in queries.iter().cloned() {
    |                  ^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_to_owned
    = note: `#[warn(clippy::unnecessary_to_owned)]` on by default
help: remove any references to the binding
    |
506 ~         for q in queries {
507 |             let client = self.clone();
508 |             let out = output_dir.clone();
509 ~             let handle = tokio::spawn(async move { client.download(q, out).await });
    |

warning: you should consider adding a `Default` implementation for `DeviceBuilder`
   --> qsib-client/src/query.rs:154:5
    |
154 | /     pub fn new() -> Self {
155 | |         Self {
156 | |             name: None,
157 | |             data_streams: Vec::new(),
...   |
160 | |     }
    | |_____^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default
    = note: `#[warn(clippy::new_without_default)]` on by default
help: try adding this
    |
152 + impl Default for DeviceBuilder {
153 +     fn default() -> Self {
154 +         Self::new()
155 +     }
156 + }
    |

warning: you should consider adding a `Default` implementation for `QueryBuilder`
   --> qsib-client/src/query.rs:269:5
    |
269 | /     pub fn new() -> Self {
270 | |         Self {
271 | |             user: None,
272 | |             device: None,
...   |
276 | |     }
    | |_____^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default
help: try adding this
    |
268 + impl Default for QueryBuilder {
269 +     fn default() -> Self {
270 +         Self::new()
271 +     }
272 + }
    |

warning: `qsib-client` (lib) generated 6 warnings (run `cargo clippy --fix --lib -p qsib-client` to apply 4 suggestions)
    Checking qfetch v0.1.0 (/Users/jwtrueb/Desktop/workspace/qsib-fetcher/qfetch)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.18s
```

### Reproducer

Clippy converted this 

<img width="1031" height="473" alt="Image" src="https://github.com/user-attachments/assets/c5a06d3f-00b6-4492-a19e-4be85ca39541" />

to this

<img width="1024" height="477" alt="Image" src="https://github.com/user-attachments/assets/494dbd11-2301-4a81-9fd4-0a02879e7c40" />

It used to move a `Query` into an async closure. After the fix, it moves a `&Query` into the async closure hence the compiler error. I figure this may have happened because the function does take a Query by reference, but it was intentionally moved into the closure then referenced to avoid this ownership issue.

### Version

```text
rustc 1.89.0 (29483883e 2025-08-04)
binary: rustc
commit-hash: 29483883eed69d5fb4db01964cdf2af4d86e9cb2
commit-date: 2025-08-04
host: aarch64-apple-darwin
release: 1.89.0
LLVM version: 20.1.7
```
